packages/gasket-plugin-docusaurus/lib/generate-default-config.js

```diff
@@ -5,10 +5,17 @@ const defaultConfig = path.join(__dirname, '..', 'generator', 'docusaurus.config
 /**
  * Generate base docusaurus config
  *
- * @param {string} name - Name of the Gasket app - defines meta info in the config
+ * @param {object} configOptions - Object to contain Docusaurus config options that will be added to the root config
  * @returns {string} - docusaurus config file
  */
-module.exports = async function generateDefaultConfig(name) {
-  const configStr = await readFile(defaultConfig, 'utf-8');
-  return configStr.replace(/(\$\{name\})/g, name);
+module.exports = async function generateDefaultConfig(configOptions) {
+  const configProperties = Object.keys(configOptions);
+  let configStr = await readFile(defaultConfig, 'utf-8');
+
+  configProperties.forEach(property => {
+    const regex = `\${${property}}`.replace(/[${}]/g, '\\$&');
+    configStr = configStr.replace(new RegExp(regex, 'g'), configOptions[property]);
```

Comment:
I'm not sure exactly what line 16 is supposed to accomplish, but I think I know what you're going for. However, instead of iterating over the config properties, it would be simpler code to iterate of the matches with a [replacer function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_function_as_a_parameter), _then_ get the corresponding config option.

Something like:
```js
const configStr = await readFile(defaultConfig, 'utf-8');
return configStr.replace(/\$\{(\w+)\}/g, (full, propName) => {
    return configOptions[propName];
})
```